build_tools/cmake/iree_run_module_test.cmake

```diff
@@ -0,0 +1,222 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# Utility function to check expected failure for different platforms.
+function(platform_xfail OUTPUT_XFAIL XFAIL_ARGS)
+  set(${OUTPUT_XFAIL} FALSE PARENT_SCOPE)
+  if(XFAIL_ARGS)
+    if((CMAKE_SYSTEM_PROCESSOR STREQUAL "riscv64" AND
+        CMAKE_SYSTEM_NAME STREQUAL "Linux" AND
+        XFAIL_ARGS MATCHES "rv64") OR
+       (ANDROID AND XFAIL_ARGS MATCHES "arm64-v8a") OR
+       (XFAIL_ARGS MATCHES "x86_64"))
+      set(${OUTPUT_XFAIL} TRUE PARENT_SCOPE)
+    endif()
+  endif()
+endfunction()
+
+
+# iree_run_module_test()
+#
+# Creates a test using iree-run-module to run an IREE module (vmfb).
+# The function supports cross compile, so there is no matching bzl rule.
+#
+# Parameters:
+#   NAME: Name of the target
+#   MODULE_SRC: IREE module (vmfb) file.
+#   BENCHMARK_MODULE_SRC: IREE module from benchmark_suite.
+#   DRIVER: Driver to run the module with. This can be overridden if it is
+#       provided by the benchmark_suite's flagfile.
```

Comment:
Does the flagfile ever *not* provide the driver? It seems weird to have this here and have it get silently overridden

Reply:
I changed the silent override to explicit consistency check.